chariots/core/base_op.py

```diff
@@ -64,7 +71,11 @@ def _map_op(self, data_set: DataSet):
         return DataSet.from_op(map(self._perform_single, data_set))
     
     def _perform_single(self, data: DataBatch):
-        return {self.name: self._main(**data)}
+        args_dict = {arg_name: next(data_batch for data_marker, data_batch in data.items() if data_marker.compatible(marker))
```

Comment:
replace `data_marker.compatible(marker)` by `marker.compatible(data_marker)` as the compatibility is not a commutative property: an embedding might be compatible for a matrix but all matrices are not compatible for an embedding